### ISSUE TITLE
can't bottle a package with non-SemVer versioning

### ISSUE THREAD

**Issue by ddobrinskiy** [teaxyz/cli#380]
This issue is a continuation of [this discussion on discord](https://discordapp.com/channels/906608167901876256/1074748881230905407/1074748881230905407)

I'm trying to package [mamba](https://mamba.readthedocs.io/en/latest/installation.html) in [388](https://github.com/teaxyz/pantry.extra/pull/388), and it has weird version formatting i.e. `22.11.1-4`

See releases [here](https://github.com/conda-forge/miniforge/releases)

Since tea currently forces SemVer  formatting, this results in an error `Uncaught (in promise) Error: invalid version: 22.11.1-4`

Some examples of mamba versions here:

* `22.11.1-4`
* `22.11.1-3`
* `22.11.1-2`
* `22.11.1-1`
* `22.11.1-0`
* `22.9.0-3`
* `22.9.0-2`
* `22.9.0-1`
* `22.9.0-0'

So it uses a semver plus a suffix 

I wonder what's the solution here?

**Comment by mxcl** [comment]
Yeah we have a few of these. Our plans are to address them one by one. OSS is full of weird versions for sure. Though our protocol will require semver (thankfully).

For other such things currently we:

```yaml
versions:
  - 22.11.1.4
```

Then hardcode the version in the URL as `22.11.1-4`.

Our current idea for dealing with this going forwards is to allow packages to provide a custom `version.anyext` that provides a transformation into a standard dot separated form.

Sorry for the lack of clarity in our docs for this. We're still figuring it out.

With brew we allowed “anything” and this rapidly turned into a very messy regex file. This time round we want to ensure we are getting versions—however they occur in the wild—into a form we can understand.
